### PR TITLE
Fix ulimited body parsing

### PR DESCRIPTION
**Pull request by const-t**
If the response doesn't have body tempesta parses the body until the connection is closed, in this case need to update overall body size and don't make chunks.

-Added error handling to `tfw_cache_h2_copy_body()`

### CODE REVIEW

**Comment by const-t** [comment]
@krizhanovsky Is [this condition](https://github.com/tempesta-tech/tempesta/blob/master/fw/http.c#L1624) correct? If so `tfw_apm_hm_srv_alive()` will be called only for suspended server.

**Review by krizhanovsky — APPROVED** [review]
I have only one question and if there is no problem, then just comment why do we exepct body in the first skb and merge.

Regarding `tfw_http_hm_control()` - we return from the function if the server is alive (not suspended) since we do not need to mark it alive with `tfw_srv_mark_alive()`, so the code seems correct. This is an example why it's bad to write code without comments :)

File: fw/http.h [hunk 1]
```diff
@@ -713,6 +706,78 @@ tfw_http_msg_header_table_size(void)
...
+		   struct sk_buff* start, struct sk_buff* end)
+{
+	int r;
+
+	it->skb_head = end;
+	it->skb = start;
+	it->frag = -1;
+
+	/* Set starting position. */
+	r = ss_skb_find_frag_by_offset(it->skb, body_start, &it->frag);
```
[the remark this answers is not in the document]
Comment (const-t) [reply fw/http.h:730]: This is not first skb. `tfw_body_iter_init()` - Initializes iterator with data obtained during body parsing, we pass to this function pointer to start of the body and pointer to `sk_buff` that holds the body or first chunks of the body. We use these pointers as starting point. If skb `start` doesn't "contain" passed pointer `body_start` error will be handled below. `ss_skb_find_frag_by_offset()` finds number of the fragment that contains `body_start` pointer.

**Comment by const-t** [comment]
> Regarding `tfw_http_hm_control()` - we return from the function if the server is alive (not suspended) since we do not need to mark it alive with `tfw_srv_mark_alive()`, so the code seems correct. This is an example why it's bad to write code without comments :)

This part is ok. However looks strange following condition if we take into account that `tfw_apm_hm_srv_alive()` must be called for **alive** server at [least once](https://github.com/tempesta-tech/tempesta/wiki/Health-monitor) to calculate valid crc32 in case when directive `resp_crc32` is `auto`:
```
if (!tfw_srv_suspended(srv) ||
	    !tfw_apm_hm_srv_alive(resp->status, &resp->body, srv->apmref))
```

File: fw/apm.c [hunk 2]
```diff
@@ -1260,14 +1275,22 @@ tfw_apm_hm_srv_alive(int status, TfwStr *body, void *apmref)
...
+			  hm->name, status);
+		return false;
+	}
+
 	/*
 	 * Special case for 'auto' monitor: generate crc32
 	 * from body of first response and store it into monitor.
 	 */
 	if (!hm->crc32 && hm->auto_crc) {
-		hm->crc32 = tfw_str_crc32_calc(body);
```
Comment (EvgeniiMekhanik) [inline fw/apm.c:1268]: tfw_str_crc32_calc is now unused, we can remove it from our code and tests

**Review by EvgeniiMekhanik — APPROVED** [review]

File: fw/apm.c (hunk 2 above)
Comment (const-t) [reply fw/apm.c:1268]: I would prefer don't remove this function. Time to time we use crc32 for `TfwStr` even for debug purposes and it useful to have such function.

**Pull request merged**